lib/hash.ts

```diff
@@ -0,0 +1,57 @@
+// This module implements the "double hash" signature specification described in docs/signature_specification.md
+import * as crypto from 'crypto';
+import { FileContent, FilePath, FullFileHash, SignatureResult } from './types';
+import { isBinary } from './utils/binary';
+import { removeWhitespaces } from './utils/format';
+
+enum DigestFormat {
+  BASE64 = 'base64',
+  HEX = 'hex',
+}
+
+enum HashAlgorithm {
+  MD5 = 'md5',
+  OTHER = 'other',
+}
+
+const usedHashAlgorithm: HashAlgorithm = HashAlgorithm.MD5;
+
+export async function getHashSignature(
+  path: FilePath,
+  content: FileContent,
+): Promise<SignatureResult> {
+  const hashes: FullFileHash[] = await Promise.all([
+    computeHash(content),
+    computeUHash(content),
+  ]);
+
+  return {
+    path: path,
+    size: content.length,
+    hashes_ffm: hashes,
+  };
+}
+
+export async function computeHash(content: FileContent): Promise<FullFileHash> {
```

Comment:
Indeed, the name of the function is not properly expressing the logic behind. 'DoubleHash' I think also might be misleading because the actual implementation will just do a single hash. What about naming it `computeSingleHash`, do you agree?

Reply:
Resolving as function's been renamed